README.md
![alt Contenta][logo]

# contenta
## 'Things Contained'

>	content 2 |ˈkänˌtent| _noun_

>	ORIGIN late Middle English: from medieval Latin contentum (plural **contenta** ‘_things contained_’)

This project began for several reasons.

* As a Java (backend) and iOS developer I was a little frustrated with the complexity required to build a full Java backend to support a prototype iOS application.  There _must_ be a simple way to spin up some basic database driven services with basic pages and RESTful API
* Yes there are _tons_ of tools that would do what I wanted, but ..
** I need a tool I could deploy with an external face
** I need a self contained database (sqlite?) because my IT department can't support their own asses let alone a series of Oracle or MySQL or MariaDB instances
** It's fun to learn new tools
** Since it's just for me, I'm going to reinvent the wheel :)  .. again.

So this is a learning project and I have always found a _practical_ learning project teaches the most.  In other words it needs to actually solve a need for me.  Well it just so happens I have a need.  [Mylar](https://github.com/evilhero/mylar) is an interesting project, but _python_?  Based on [Headphones](https://github.com/rembo10/headphones) but .. Mylar keeps crashing on my Synology diskstation. So why not rewrite Mylar with my own requirements .. but not _python_.

Fun .. right?  So pick a language.  Well it needs to be deployed on my Synology .. but most things can.  I could go with Node.js, or PHP.  Not Java, I do enough of that at work and this is for _learning_.  I even toyed with [Swift](https://developer.apple.com/swift/) but then I would need a Mac (MacMini?  Old Laptop?) to deploy.  After testing with Node.js and PHP I decided to go with PHP.  I know it's a little old, but it has the advantage of deploying as part of a simple Apache site.

So now, what are the requirements?  I told you this would be fun.

1. Multi-user support
1. upload comics and have them automatically organized and indexed using [ComicVine][comicvine]
1. track what I've read
1. content rating?  My twins are only 7, but they might enjoy the library soon.
1. integrate with SabNZBd and automatically download content
1. load NZB RSS feeds
1. search Newznab based sites for matching content
1. target CBR/CBZ as a content type, but leave room for adding PDF/ePub or other formats
1. hmm, actually lets unpack the CBR and store them as CBZ .. I dislike RAR.  It's personal.
1. provide a simple RESTful api for scripting
1. although I don't desire to write an iOS comic book reader, maybe an iOS _Admin_ application ...

Thats enough to start ..

[logo]: https://raw.githubusercontent.com/vitolibrarius/contenta/master/public/img/Logo_sm.png "Contenta"
[comicvine]: http://comicvine.gamespot.com

- vito


CHANGES
Version 0.9.1:
* 29a5fd3 -  (HEAD -> master) Added ability to specify Comic Vine volume for import (Sun Feb 10 16:56:10 2019 -0700) <Vito Librarius>

Version 0.9.0:


Version 0.8.1:
* 0718d78 -  (HEAD -> master) Added Daily maximum request counter (Wed Aug 16 22:21:08 2017 -0600) <Vito Librarius>
* 8b60f0b -  (origin/master, origin/HEAD) Changed response code to 409 for already imported content (Fri Jul 28 11:20:02 2017 -0600) <Vito Librarius>
* ad7dff1 -  Issue #82: comicvine seems to always want the api_key as the first parameter (Fri May 26 09:38:10 2017 -0600) <Vito Librarius>
* 3122324 -  Issue 82: artist roles can be comma separate (Thu May 25 08:45:07 2017 -0600) <Vito Librarius>

Version 0.8.0:
* 309177d -  (HEAD -> master, origin/master, origin/HEAD) Fixes #151: added display of artists grouped by roles (Wed May 24 21:49:15 2017 -0600) <Vito Librarius>
* ff81835 -  Issue #151: modeled Artists and relationships to Pubs/Series.  Updated importer to load Artist (Wed May 24 11:30:31 2017 -0600) <Vito Librarius>
* 46a1914 -  Issue #82: adjusted endpoint error reporting (Mon May 15 14:18:34 2017 -0600) <Vito Librarius>
* ab5d993 -  Issue #82: modified search priority order (Mon May 15 12:05:16 2017 -0600) <Vito Librarius>
* c6d686b -  Fixes #81: better feedback to processes (Fri May 12 21:34:27 2017 -0600) <Vito Librarius>
* 2b7bb2d -  Issue #82: adjusted intervals for nzb searching (Fri May 12 17:19:54 2017 -0600) <Vito Librarius>
* 361850a -  Issue #82: fixed non-object error (Fri May 12 10:41:46 2017 -0600) <Vito Librarius>
* 3ce3d9e -  Fixes #150: added publication cards (Thu May 11 21:04:22 2017 -0600) <Vito Librarius>

Version 0.7.5:
*   33707eb -  (HEAD -> master, origin/master, origin/HEAD) Merge branch 'issue-149-automated-search-should-prioritize-recent-publicat' (Wed May 10 22:51:06 2017 -0600) <Vito Librarius>
|\  
| * 9419a91 -  Fixes #149: tracking last search date (Wed May 10 22:50:34 2017 -0600) <Vito Librarius>
* | 01881c6 -  Issue 82: fewer exceptions in newznab connection (Mon May 8 09:23:33 2017 -0600) <Vito Librarius>
* | e1d30f0 -  Issue #82 Rss connector should not throw exception due to no data (Mon May 8 09:14:30 2017 -0600) <Vito Librarius>
|/  
* 0faca37 -  Issue #82: Cast code to int (Fri May 5 10:13:30 2017 -0600) <Vito Librarius>
* 2f6b711 -  Issue #82: allow flux to retry (Fri May 5 08:04:59 2017 -0600) <Vito Librarius>
* 02f3e29 -  issue #82: modified endpoint connection error logs (Thu May 4 22:48:05 2017 -0600) <Vito Librarius>
* fe9fc5f -  Issue #82: search size file warning (Wed May 3 22:35:42 2017 -0600) <Vito Librarius>
* 7885675 -  Fixes #57: improved parsing and display of newznab result data (Wed May 3 22:04:05 2017 -0600) <Vito Librarius>
* ce6ed1c -  Fixes #111: allows nzb downloads to be retried (Wed May 3 21:55:22 2017 -0600) <Vito Librarius>
* a62dc82 -  Issue #82 trying to trace silent error on quickNewznab search function (Wed May 3 10:28:17 2017 -0600) <Vito Librarius>
* 74d6837 -  Issue #82: increased JS tiemouts for typing to 1000.  Added media size search (Fri Apr 28 13:36:50 2017 -0600) <Vito Librarius>
* e42b3b0 -  Issue #82: updated layout (Wed Feb 22 09:33:53 2017 -0700) <Vito Librarius>
* 9ae8279 -  Issue #82 improved progress displays (Tue Feb 21 22:02:35 2017 -0700) <Vito Librarius>
* 64bf91b -  Issue #82: check larger sabnzb history slots (Tue Feb 14 21:26:00 2017 -0700) <Vito Librarius>
* 2a94858 -  Issue #82: Default for Story Arcs is to not make them wanted (Tue Feb 14 17:23:05 2017 -0700) <Vito Librarius>
* 1606176 -  Issue #82: tweaking the css (Tue Feb 14 10:13:19 2017 -0700) <Vito Librarius>
* 86f51b4 -  Update reading item when slideshow (Fri Feb 3 17:53:49 2017 -0700) <Vito Librarius>
* 13d26d4 -  restored reading list (Fri Feb 3 17:51:04 2017 -0700) <Vito Librarius>
*   ea53837 -  Merge pull request #148 from vitolibrarius/angular (Fri Feb 3 17:05:28 2017 -0700) <Vito Librarius>
|\  
| * 93f3d91 -  (origin/angular, angular) Fixes #105 slideshow (Fri Feb 3 17:00:06 2017 -0700) <Vito Librarius>
| * 33645a6 -  experimenting (Tue Jan 17 17:00:42 2017 -0700) <Vito Librarius>
| * c8f365f -  experimental menu components/module/controller (Fri Jan 13 09:25:26 2017 -0700) <Vito Librarius>
| * 6df22d7 -  Initial changes to get an angular test but working (Thu Jan 12 20:57:41 2017 -0700) <Vito Librarius>
|/  
* 5f71f6f -  Issue #82 updated headers and footers (Thu Jan 12 20:56:27 2017 -0700) <Vito Librarius>
* 158d44d -  (issue-143-exclusions-and-expansions-need-an-editor-interface) Issue #82: Wanted StoryArc failed to update after nzb requested (Tue Jan 10 11:40:40 2017 -0700) <Vito Librarius>
* d0dc160 -  Fixes #36	Publisher list needs paging or searching (Tue Jan 10 10:59:19 2017 -0700) <Vito Librarius>
* 49c9801 -  Fixes #144	Logs: better search / paging / delete functions (Mon Jan 9 21:30:34 2017 -0700) <Vito Librarius>
* 681d9cc -  Issue #82 - shutdown handler should only log E_ERROR events (Sun Jan 8 11:37:40 2017 -0700) <Vito Librarius>
* 0e1d1df -  Issue #82 misc (Fri Dec 16 17:37:01 2016 -0700) <Vito Librarius>
* aacb23f -  Fixes #147	Exception: libs/connectors/EndpointConnector.php (Fri Dec 16 16:52:31 2016 -0700) <Vito Librarius>
* e5796a9 -  (issue-146-errors-modelpulllistpulllistexclusiondbophp) Fixes #145	application/config/common_images.php - exception when no image found (Fri Dec 16 16:43:14 2016 -0700) <Vito Librarius>
* d5cbef3 -  (issue-142-rss-feeds-should-have-expansion-and-exclusions-lik, issue-135-add-series-and-story-ratings) Issue #82 small fixes (Fri Dec 16 15:29:11 2016 -0700) <Vito Librarius>
* b31d0e1 -  Issue #82: page params are not initialized in all cases, added null check (Wed Dec 7 20:45:32 2016 -0700) <Vito Librarius>
* 2e8f2a0 -  Issue #82: improved word wrap for URLs (Fri Nov 25 17:58:41 2016 -0700) <Vito Librarius>
* bad47dd -  Issue #82: changed sort order for pull list item (Thu Nov 10 18:53:09 2016 -0700) <Vito Librarius>

phpunit/CommonTest.php
<?php
/**
 * DO NOT EDIT CUSTOM MARKERS
 * Generated from Class.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
 */



use \PHPUnit_Framework_TestCase as PHPUnit_Framework_TestCase;
/* {useStatements} */

class CommonTest extends PHPUnit_Framework_TestCase
{
    protected function setUp()
    {
    }

    protected function tearDown()
    {
    }

    public function sanitationFilenames()
    {
		$metadata = test_jsonResource("SanitationFilenames.json");
		return $metadata->getMeta( "/" );
    }

    public function sanitationStrings()
    {
		$metadata = test_jsonResource("SanitationStrings.json");
		return $metadata->getMeta( "/" );
    }

    public function normalizationStrings()
    {
		$metadata = test_jsonResource("NormalizedStrings.json");
		return $metadata->getMeta( "/" );
    }

/*	 Test functions */

	/**
	 * @covers	callerClassAndMethod
	 * 			T_FUNCTION callerClassAndMethod ( $currentFunction = '')
	 * @todo	Implement testCallerClassAndMethod().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testCallerClassAndMethod()
	{
		$caller = callerClassAndMethod();
		$this->assertEquals( 'CommonTest', $caller['class'] );
		$this->assertEquals( 'testCallerClassAndMethod', $caller['function'] );

		$caller = callerClassAndMethod('testCallerClassAndMethod');
		$this->assertEquals( 'ReflectionMethod', $caller['class'] );
		$this->assertEquals( 'invokeArgs', $caller['function'] );
	}

	/**
	 * @covers	split_lines
	 * 			T_FUNCTION split_lines ( $str)
	 * @todo	Implement testSplit_lines().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testSplit_lines()
	{
		$lines = split_lines( null );
		$this->assertCount( 0, $lines );

		$lines = split_lines( "one line no breaks" );
		$this->assertCount( 1, $lines );

		$text = "one" .  PHP_EOL . "two\rthree\rfour\n\rfive";
		$lines = split_lines( $text );
		$this->assertCount( 5, $lines );
	}

	/**
	 * @covers	get_short_class
	 * 			T_FUNCTION get_short_class ( $obj)
	 * @todo	Implement testGet_short_class().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testGet_short_class()
	{
		$this->assertEquals( 'CommonTest', get_short_class($this) );
	}

	/**
	 * @covers	startsWith
	 * 			T_FUNCTION startsWith ( $needle, $haystack)
	 * @todo	Implement testStartsWith().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testStartsWith()
	{
		$this->assertFalse( startsWith(null, "Vitolibrarious"));
		$this->assertFalse( startsWith("vito", null));
		$this->assertFalse( startsWith(null, null));
		$this->assertFalse( startsWith("vito", "Vitolibrarious"));

		$this->assertTrue( startsWith("vito", "vitolibrarious"));
	}

	/**
	 * @covers	endsWith
	 * 			T_FUNCTION endsWith ( $needle, $haystack)
	 * @todo	Implement testEndsWith().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testEndsWith()
	{
		$this->assertFalse( endsWith(null, "Vitolibrarious"));
		$this->assertFalse( endsWith("vito", null));
		$this->assertFalse( endsWith(null, null));
		$this->assertFalse( endsWith("librarious", "VitoLibrarious"));

		$this->assertTrue( endsWith("librarious", "vitolibrarious"));
	}

	/**
	 * @covers	contains
	 * 			T_FUNCTION contains ( $haystack, $needle)
	 * @todo	Implement testContains().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testContains()
	{
		$this->assertFalse( contains(null, "Vitolibrarious"));
		$this->assertFalse( contains("vito", null));
		$this->assertFalse( contains(null, null));
		$this->assertFalse( contains("lib", "VitoLibrarious"));

		$this->assertTrue( contains("lib", "vitolibrarious"));
		$this->assertTrue( contains("vito", "vitolibrarious"));
		$this->assertTrue( contains("rious", "vitolibrarious"));
	}

	/**
	 * @covers	zipFileList
	 * 			T_FUNCTION zipFileList ( $path)
	 * @todo	Implement testZipFileList().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testZipFileList()
	{
		$zipfile = test_mediaSamplesFile( null );
		$filelist = zipFileList($zipfile);
		$this->assertFalse( is_array($filelist) );

		$zipfile = test_mediaSamplesFile( 'not a file' );
		$filelist = zipFileList($zipfile);
		$this->assertFalse( is_array($filelist) );

		$zipfile = test_mediaSamplesFile( "Space_Wikimedia.cbz" );
		$filelist = zipFileList($zipfile);
		$this->assertTrue( is_array($filelist) );
		$this->assertCount( 4, $filelist );
	}

	/**
	 * @covers	classNames
	 * 			T_FUNCTION classNames ( $file)
	 * @todo	Implement testClassNames().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testClassNames()
	{
		$classes = classNames( null );
		$this->assertFalse( is_array($classes) );

		$classes = classNames( "/not/a/real/path" );
		$this->assertFalse( is_array($classes) );

		$zipfile = test_mediaSamplesFile( "Space_Wikimedia.cbz" );
		$classes = classNames( $zipfile );
		$this->assertTrue( is_array($classes) );
		$this->assertCount( 0, $classes );

		$classes = classNames( __FILE__ );
		$this->assertTrue( is_array($classes) );
		$this->assertCount( 1, $classes );
	}

	/**
	 * @covers	uuid
	 * 			T_FUNCTION uuid ( )
	 * @todo	Implement testUuid().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testUuid()
	{
		$uuid = uuid();
		$this->assertTrue( is_string($uuid) );
		$this->assertEquals( 36, strlen($uuid));
	}

	/**
	 * @covers	uuidShort
	 * 			T_FUNCTION uuidShort ( )
	 * @todo	Implement testUuidShort().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testUuidShort()
	{
		$uuid = uuidShort();
		$this->assertTrue( is_string($uuid) );
		$this->assertEquals( 32, strlen($uuid));
	}

	/**
	 * @covers	sanitize_filename
	 * 			T_FUNCTION sanitize_filename ( $string, $maxLength, $force_lowercase = true, $anal = false)
     * @dataProvider sanitationFilenames
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testSanitize_filename( $teststring, $tests )
	{
		foreach( $tests as $type => $expected ) {
			list( $name, $length, $lowercase, $anal ) = explode(":", $type);

			// Returns TRUE for "1", "true", "on" and "yes"
			// Returns FALSE for "0", "false", "off" and "no"
			// Returns NULL otherwise.
			$lowercase = filter_var($lowercase, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);
			$anal = filter_var($anal, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);

			$sane = sanitize_filename( $teststring, intval($length), $lowercase, $anal);
			$this->assertTrue( strlen($sane) <= $length, "String length" );
			$this->assertEquals( $expected, $sane, "Using rules $type = " . intval($length)
				.":". var_export($lowercase, true) .":". var_export($anal, true) );
		}
	}

	/**
	 * @covers	sanitize
	 * 			T_FUNCTION sanitize ( $string, $force_lowercase = true, $anal = false)
     * @dataProvider sanitationStrings
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testSanitize( $teststring, $tests)
	{
		foreach( $tests as $type => $expected ) {
			list( $name, $lowercase, $anal ) = explode(":", $type);

			// Returns TRUE for "1", "true", "on" and "yes"
			// Returns FALSE for "0", "false", "off" and "no"
			// Returns NULL otherwise.
			$lowercase = filter_var($lowercase, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);
			$anal = filter_var($anal, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);

			$sane = sanitize( $teststring, $lowercase, $anal);
			$this->assertEquals( $expected, $sane, "Using rule $type ("
				. var_export($lowercase, true) .":". var_export($anal, true) . ")" );
		}
	}

	/**
	 * @covers	normalize
	 * 			T_FUNCTION normalize ( $string)
     * @dataProvider normalizationStrings
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testNormalize($teststring, $normal, $normal_search)
	{
		$new_normal = normalize( $teststring );
		$this->assertEquals( $new_normal, $normal );
	}

	/**
	 * @covers	normalizeSearchString
	 * 			T_FUNCTION normalizeSearchString ( $string = null)
     * @dataProvider normalizationStrings
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testNormalizeSearchString($teststring, $normal, $normal_search)
	{
		$new_normal = normalizeSearchString( $teststring );
		$this->assertEquals( $new_normal, $normal_search );
	}

	/**
	 * @covers	words
	 * 			T_FUNCTION words ( $string)
	 * @todo	Implement testWords().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testWords()
	{
		$this->markTestIncomplete('This test has not been implemented yet.');
	}

	/**
	 * @covers	lines
	 * 			T_FUNCTION lines ( $string)
	 * @todo	Implement testLines().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testLines()
	{
		$lines = lines();
		$this->assertNull( $lines, "null" );

		$lines = lines('');
		$this->assertNull( $lines, "empty string" );

		$lines = lines("\n\n\n\n");
		$this->assertNotNull( $lines, "newlines only" );
		$this->assertCount( 0, $lines, "newlines only, no results" );

		$lines = lines("\n\r\n\r\r\n\n");
		$this->assertNotNull( $lines, "newlines only" );
		$this->assertCount( 0, $lines, "newlines only, no results" );

		$lines = lines("the\nquick brown fox junmped\n\n\nover the lazy dog");
		$this->assertNotNull( $lines, "newlines only" );
		$this->assertCount( 3, $lines, "newlines only, no results" );
	}

	/**
	 * @covers	convertToBytes
	 * 			T_FUNCTION convertToBytes ( $val)
	 * @todo	Implement testConvertToBytes().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testConvertToBytes()
	{
		$bytes = convertToBytes();
		$this->assertEquals( 0,  $bytes, "null" );

		$bytes = convertToBytes('');
		$this->assertEquals( 0, $bytes, "null" );

		$bytes = convertToBytes('123');
		$this->assertEquals( 123, $bytes, "null" );

		$bytes = convertToBytes('1048576');
		$this->assertEquals( 1048576, $bytes, "null" );

		$bytes = convertToBytes('10M');
		$this->assertEquals( 10485760, $bytes, "null" );
	}

	/**
	 * @covers	formatSizeUnits
	 * 			T_FUNCTION formatSizeUnits ( $value)
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testFormatSizeUnits()
	{
		$bytes = formatSizeUnits();
		$this->assertEquals( "unknown",  $bytes, "null" );

		$bytes = formatSizeUnits('');
		$this->assertEquals( "unknown", $bytes, "null" );

		$bytes = formatSizeUnits('0');
		$this->assertEquals( "0 bytes", $bytes, "null" );

		$bytes = formatSizeUnits('123');
		$this->assertEquals( "123 bytes", $bytes, "null" );

		$bytes = formatSizeUnits('1048576');
		$this->assertEquals( "1.00 MB", $bytes, "null" );

		$bytes = formatSizeUnits('10M');
		$this->assertEquals( "10.00 MB", $bytes, "null" );
	}

	/**
	 * @covers	formattedTimeElapsed
	 * 			T_FUNCTION formattedTimeElapsed ( $diff)
	 * @todo	Implement testFormattedTimeElapsed().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testFormattedTimeElapsed()
	{
		$bytes = formattedTimeElapsed();
		$this->assertEquals( "less than one second",  $bytes, "null" );

		$bytes = formattedTimeElapsed('');
		$this->assertEquals( "less than one second", $bytes, "null" );

		$bytes = formattedTimeElapsed('123');
		$this->assertEquals( "2 minutes", $bytes, "null" );

		$bytes = formattedTimeElapsed('1048576');
		$this->assertEquals( "1 week", $bytes, "null" );

		$bytes = formattedTimeElapsed('104857653');
		$this->assertEquals( "3 years", $bytes, "null" );
	}

	/**
	 * @covers	resize_image
	 * 			T_FUNCTION resize_image ( $sourcefile, $xmax, $ymax)
	 * @todo	Implement testResize_image().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testResize_image()
	{
		$this->markTestIncomplete('This test has not been implemented yet.');
	}

	/**
	 * @covers	Zip
	 * 			T_FUNCTION Zip ( $source, $destination)
	 * @todo	Implement testZip().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testZip()
	{
		$this->markTestIncomplete('This test has not been implemented yet.');
	}

	/**
	 * @covers	currentVersionNumber
	 * 			T_FUNCTION currentVersionNumber ( )
	 * @todo	Implement testCurrentVersionNumber().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testCurrentVersionNumber()
	{
		$this->markTestIncomplete('This test has not been implemented yet.');
	}

	/**
	 * @covers	currentVersionHash
	 * 			T_FUNCTION currentVersionHash ( )
	 * @todo	Implement testCurrentVersionHash().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testCurrentVersionHash()
	{
		$this->markTestIncomplete('This test has not been implemented yet.');
	}

	/**
	 * @covers	currentRemoteHash
	 * 			T_FUNCTION currentRemoteHash ( )
	 * @todo	Implement testCurrentRemoteHash().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testCurrentRemoteHash()
	{
		$this->markTestIncomplete('This test has not been implemented yet.');
	}

	/**
	 * @covers	currentChangeLog
	 * 			T_FUNCTION currentChangeLog ( )
	 * @todo	Implement testCurrentChangeLog().
	 * Generated from Function.tpl by PhpTestClassGenerator.php on 2016-05-16 20:30:12.
	 */
	public function testCurrentChangeLog()
	{
		$this->markTestIncomplete('This test has not been implemented yet.');
	}


/* {functions} */
}
